js/stores/message_store.js

```diff
@@ -22,18 +24,39 @@ var MessageStore = assign({}, EventEmitter.prototype, {
   all: function() {
     return _messages;
   },
+
```

Comment:
So theres a fundamental problem with a) linking messages to users only with a name, b) allowing users to easily change their name but not change it for previous messages and c) allowing users to change the last message they typed. If someone changes their name I can then change *my* name to their old name and edit their message. So I'm solving this problem here by tracking which messages are local... but its worth thinking about better solutions.